src/qibolab/maps.py

```diff
@@ -0,0 +1,31 @@
+NameId = str
+
+
+class NamedType:
+    """Arbitrary object that has a name."""
+
+    name: NameId
+
+
+class NamedMap(dict):
+    """Collection of ``NamedType`` objects in a dictionary where the key is their ``name``."""
+
+    Type = NamedType
+
+    def _add_items(self, *items):
```

Comment:
A Python-closer alternative could be
```suggestion
    def extend(self, *items):
```